andela/ah-infinity-stones
# #163368898 Add email to Article's Author Details

#### What does this PR do?
- Add email to article's author details
#### Description of Task to be completed?
- Modify the article models to also include author's email
#### How should this be manually tested?
1. `git clone -b ch-add-email-to-authors-data-163368898 https://github.com/andela/ah-infinity-stones` and `$ cd ah-infinity-stones`
2. Create and activate virtual env
- `virtualenv  -p python3 venv`
- `$ source venv/bin/activate`
3. `$ pip3 install -r requirements.txt` to install the dependencies
4. Setup environment variables. Copy all from below and paste them into the .env file. Update the values to your choice.
```export DATABASE_NAME='ah_dev'
export DATABASE_USER='user'
export DATABASE_PASSWORD=pass'
export HOST='localhost'
export PORT='5432'
export DATABASE_TEST='ah_test'
export EMAIL_HOST_USER='username'
export EMAIL_HOST_PASSWORD='pass'
export SECRET_KEY='key-is-secret'
```
Activate the env variables by `$source .env`.
5. Run migrations: `$ python3 manage.py migrate`
6. Run app - `$ python3 manage.py runserver`
7. Access `http://127.0.0.1:8000` through postman and test by:
-  `POST /api/users/` - Create account with AH
Check your email and activate the account. Login with the endpoint below:
- `POST /api/users/login` - Login to AH
- Use the token generated on login to create an article `POST /api/articles`
```
{
    "title": "How to make it rain",
    "description": "Ever wonder how?",
    "body": "You have to pray",
    "tag": ["rain"]
  
}
```
- Note the article slug generated after successfully creating an article. In this case, it would be `how-to-make-it-rain`. Access the specific article by `GET api/articles/how-to-make-it-rain`
- You should see the author's email.

#### What are the relevant pivotal tracker stories?
[#163368898](https://www.pivotaltracker.com/story/show/163368898)
#### Screenshots
<img width="1433" alt="screenshot 2019-01-21 at 21 04 07" src="https://user-images.githubusercontent.com/16904946/51491716-3fa1b780-1dc0-11e9-9274-a6d9a78160af.png">
